packages/interpolate-components/package.json

```diff
@@ -22,6 +22,7 @@
 	"homepage": "https://github.com/Automattic/wp-calypso/tree/HEAD/packages/interpolate-components#readme",
 	"devDependencies": {
 		"@automattic/calypso-typescript-config": "workspace:^",
+		"@types/react": "^17.0.37",
```

Comment:
I think `@types/*` dependencies are better as `dependencies` (right @sirreal ?)

Reply:
`@types/*` dependencies should mirror the dependency they provide types for. If `react` is a `dependency`, its types should also be a `dependency`. In this case, `react` is a _peer dependency_, so **`@types/react` should be a `_peer dependency_**.

**We should also match major versions.** In this case we're depending on `react@>=16.2.0`, the types should be similar. [It looks like the 16.1 types include the definition for `Fragment`](https://unpkg.com/browse/@types/react@16.1.0/index.d.ts), which was the requirement for interpolate components. I'd recommend using `@types/react@>=16.1.0` or `16.3.0` (there's no `16.2`). As far as I can tell `Fragment` was added somewhere between 16.0 and 16.1 in the types.

---

The reasoning for all this is that the types we'll be publishing include `JSX.Element`, so any consumers will see that. If they don't have a definition for `JSX.Element` is, they'll experience errors.

We might also [mark the dependency as optional](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta). Folks using TypeScript need the dependency for the type system to work, but folks not using TypeScript have absolutely no need for `@types/react`.

Something like the following seems ideal to me:

```json
{
	"peerDependencies": {
		"react": ">=16.2.0",
		"@types/react": ">=16.1.0"
	},
	"peerDependenciesMeta": {
		"@types/react": {
			"optional": true
		}
	},
}
```

Reply:
Thanks @sirreal. I applied your suggestion. What's the publishing process @scinos? Thank you!

Reply:
The process I follow is https://github.com/Automattic/wp-calypso/blob/trunk/docs/monorepo.md#publishing